django/db/backends/oracle/introspection.py

```diff
@@ -17,39 +20,42 @@ class DatabaseIntrospection(BaseDatabaseIntrospection):
     # Maps type objects to Django Field types.
     @cached_property
     def data_types_reverse(self):
-        if self.connection.cx_oracle_version < (8,):
+        # Legacy support for cx_Oracle 7.0
+        if self.connection.oracledb_version >= (
+            7,
+        ) and self.connection.oracledb_version < (8,):
```

Comment:
Suggested formatting from black.

Reply:
I'd add `HAS_CX_ORACLE` flag to use it here:
```python
try:
    import oracledb
except ImportError:
    import cx_Oracle as oracledb
    HAS_CX_ORACLE = True
else:
    HAS_CX_ORACLE = False

...

   @cached_property
    def data_types_reverse(self):
        # RemovedInDjango51Warning.
        if HAS_CX_ORACLE and self.connection.cx_oracle_version < (8,):
```